sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -191,7 +254,56 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
       ManagedTable
     }
 
-    client.createTable(
+    val hiveSerDe = HiveSerDe.sourceToSerDe(provider, hive.hiveconf, false).getOrElse(null)
```

Comment:
Please resort to `Option[T]` combinators rather than falling back to `null` when possible. Unless we are on performance critical code paths.